test/spec/unit/pbjs_api_spec.js

```diff
@@ -2166,14 +2166,47 @@ describe('Unit: Prebid Module', function () {
         $$PREBID_GLOBAL$$.requestBids({adUnits});
         const spyArgs = adapterManager.callBids.getCall(0);
         const nativeRequest = spyArgs.args[1][0].bids[0].nativeParams;
-        expect(nativeRequest).to.deep.equal({
-          image: {required: true},
-          title: {required: true},
-          sponsoredBy: {required: true},
-          clickUrl: {required: true},
-          body: {required: false},
-          icon: {required: false},
-        });
+        expect(nativeRequest.ortb.assets).to.deep.equal([
```

Comment:
I don't think they should be mutually exclusive. Thousands of publishers already have native ad units set up in the old way, but in the future bidders will start to switch to ortb2, so for some time the same ad unit will contain both types.  

Then, it's the bidder's responsibility to decide which version they want to use. I think that ortb will be the preferred format since it's more expressive, but waiting for bidders to migrate can take ages. That's why we've also given the two functions to convert to and from legacy prebid native format. 

 @bretg wdyt ?

Reply:
But a publisher is expected to specify either one or the other, right? otherwise it's not clear how you'd "merge" the definitions, or what to do in case they conflict. That's what I meant by validation.

Reply:
I think publishers may specify both types in the same ad unit. This shouldn't be seen as an error. It's the single adapter that will have to adapt its code to both formats. 

Right now publishers are using only the proprietary format. When this PR is merged, the vast majority of these publishers will continue to use the proprietary format. Then, a bunch of native bid adapters will start to support the ortb format and will ask publishers to bid in this new format. So we'll have the situation where, for the same ad unit, some native bidders will understand the old way _only_, others will understand both ways, and some others will understand the new way only. It's up to them to decide how to interpret the assets object. 

If I were a bidder, i'd implement this logic: 
```
let ortb; 
if (nativeType.native && nativeType.native.ortb) {
   // use only ortb
   {ortb} = nativeType.native; 
} else {
   // publisher is using old proprietary format, it's up to them to 
   // decide wether use it straight or convert to ortb with function 
   // `toOrtbNative()` 
   ortb = toOrtbNative(nativeType.native)
}
```

Reply:
Wouldn't it be better for a publisher to switch over to ortb, and let the translation shim you already added in here handle the adapters that have not yet been ported? With the goal of having every bid adapter talk only ortb at some point in the future.  Having to maintain two versions would add a lot of friction and I don't see the need for it.

Reply:
To clarify, I see the transition as:

- adapters initially understand the old way only (and this PR makes sure they get it regardless of how the adunit is defined)
- gradually, adapters will understand the new way only (and they can translate old definitions as you show above)

But I don't think publishers or adapters should be interested in mixing or matching the formats; it's going to invite inconsistencies otherwise.

Reply:
@dgirardi added the code to check if ortb is in use, no legacy native keys should be used; this has been done [here](https://github.com/prebid/Prebid.js/pull/8086/files#diff-f2f8b0820ed4ec1bd15383b439def39eb86117e358d9a14eaf5dc1ad73439dcaR132) and it's called by `checkAdUnitSetup` in both prebid and sizeMappingV2